### DISCUSSION TITLE
Export CSV does not export Lens fields

### DISCUSSION THREAD

**Discussion by morrislaptop** [laravel/nova-issues#4264] [Q&A] [answered]
- Laravel Version: 9.x
- Nova Version: 4.x
- PHP Version: 8.x
- Database Driver & Version: Postgres 12
- Operating System and Version: Mac 12.3
- Browser type and version: Chrome 101
- Reproduction Repository: ...

**Comment by crynobone** [comment]
The Export CSV uses database fields, you should add `withFormat()` 

<img width="782" alt="Screenshot 2022-05-16 at 6 50 34 AM" src="https://user-images.githubusercontent.com/172966/168497322-5275c89b-3e71-474b-9313-cafc75b7e434.png">

**Comment by morrislaptop** [comment]
@crynobone What about a Lens with a group by?

**Comment by morrislaptop** [comment]
It works with Selected Resources = all, but from the looks of it the ExportCsv class goes back down to the Index Query of the underlying resource if some resources are selected? Seems ExportCsv doesn't support Lens and is a bug?

**Reply by moisessepulveda** [reply]
i have the same issue

**Reply by benedict-w** [reply]
I'm also exporting a Lens with a GROUP BY query, and it only exports records when all resources are selected (also Laravel 9.x and Nova 4.x)

**Reply by RenoLooijmans** [reply]
I'm experiencing the same issue.

**Reply by kazak71** [reply]
I have the same problem

**Comment by crynobone** [comment, marked as answer]
Limitation with Group By: See https://github.com/laravel/nova-issues/discussions/5020
